public/app/features/dashboard/components/PanelEditor/PanelHeaderCorner.tsx

```diff
@@ -103,15 +91,86 @@ export class PanelHeaderCorner extends Component<Props> {
     }
 
     if (infoMode === InfoMode.Error && error) {
-      return this.renderCornerType(infoMode, error, this.onClickError);
+      return <PanelInfoCorner infoMode={infoMode} content={error} onClick={this.onClickError} />;
     }
 
     if (infoMode === InfoMode.Info || infoMode === InfoMode.Links) {
-      return this.renderCornerType(infoMode, this.getInfoContent);
+      return <PanelInfoCorner infoMode={infoMode} content={this.getInfoContent} />;
     }
 
     return null;
   }
 }
 
 export default PanelHeaderCorner;
+
+interface PanelInfoCornerProps {
+  infoMode: InfoMode;
+  content: PopoverContent;
+  onClick?: () => void;
+}
+
+function PanelInfoCorner({ infoMode, content, onClick }: PanelInfoCornerProps) {
+  const theme = infoMode === InfoMode.Error ? 'error' : 'info';
+  const ariaLabel = selectors.components.Panels.Panel.headerCornerInfo(infoMode.toLowerCase());
+  const styles = useStyles2(getStyles);
+
+  return (
+    <Tooltip content={content} placement="top-start" theme={theme} interactive>
+      <button type="button" className={styles.infoCorner} onClick={onClick} aria-label={ariaLabel}>
+        <Icon
+          name={iconMap[infoMode]}
+          size={infoMode === InfoMode.Links ? 'sm' : 'lg'}
+          className={cx(styles.icon, infoMode === InfoMode.Links ? styles.iconLinks : '')}
+        />
+        <span className={cx(styles.inner, infoMode === InfoMode.Error ? styles.error : '')} />
+      </button>
+    </Tooltip>
+  );
+}
+
+const iconMap: Record<InfoMode, IconName> = {
+  [InfoMode.Error]: 'exclamation',
+  [InfoMode.Info]: 'info',
+  [InfoMode.Links]: 'external-link-alt',
+};
+
+const getStyles = (theme: GrafanaTheme2) => {
+  return {
+    icon: css({
+      position: 'absolute',
+      top: 0,
+      left: 0,
+      zIndex: 3,
```

Comment:
gotta ask about the magic `zIndex` numbers

in the previous scss, i believe this resolved to 2 - why 3 now? 🤔